numpy/core/src/umath/ufunc_object.c

```diff
@@ -960,35 +976,69 @@ static int get_ufunc_arguments(PyUFuncObject *ufunc,
                      * Overrides the global parameters buffer size,
                      * error mask, and error object
                      */
-                    if (strncmp(str,"extobj",6) == 0) {
+                    if (strncmp(str, "extobj", 6) == 0) {
                         *out_extobj = value;
                         bad_arg = 0;
                     }
                     break;
                 case 'o':
-                    /* First output may be specified as a keyword parameter */
-                    if (strncmp(str,"out",3) == 0) {
-                        if (out_op[nin] != NULL) {
+                    /*
+                     * Output arrays may be specified as a keyword argument,
+                     * either as a single array (or None) for single output
+                     * ufuncs, or as a tuple of arrays (or Nones).
+                     */
+                    if (strncmp(str, "out", 3) == 0) {
+                        if (nargs > nin) {
                             PyErr_SetString(PyExc_ValueError,
                                     "cannot specify 'out' as both a "
                                     "positional and keyword argument");
                             goto fail;
                         }
-
-                        if (PyArray_Check(value)) {
-                            const char *name = "output array";
-                            PyArrayObject *value_arr = (PyArrayObject *)value;
-                            if (PyArray_FailUnlessWriteable(value_arr, name) < 0) {
+                        if (PyTuple_Check(value) &&
+                            PyTuple_GET_SIZE(value) == nout) {
+                            /* 'out' must be a tuple of arrays */
+                            for(i = 0; i < nout; ++i) {
+                                PyObject *val = PyTuple_GET_ITEM(value, i);
+                                if (_check_out_array(val, out_op+nin+i) < 0) {
+                                    goto fail;
+                                }
+                            }
+                        }
+                        else if (nout == 1) {
+                            /* Can be an array if it only has one output */
+                            if (_check_out_array(value, out_op+nin) < 0) {
```

Comment:
Not being used very much to `C` I'm happy to follow convention -- though this particular piece of code did not seem to follow it originally.  Anyway, I think that with the name change it will be much more obvious that the main action is something being *set*, not just *checked*.